From cddfef0ddefe71e04b82676733a740bf37bad020 Mon Sep 17 00:00:00 2001
From: Meticulus <theonejohnnyd@gmail.com>
Date: Thu, 7 Jun 2018 08:13:59 -0500
Subject: [PATCH] "0001-init-vendor-stubs-boot-critical
 _______________________________

*Previous From e41de8b223edc0da3df9d53ef9e2aaad4ccc6f30 Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Wed, 28 Feb 2018 09:00:28 -0600
*Previous Subject: [PATCH] "0001-init-vendor-stubs-boot-critical
*_______________________________
*
**Previous From f5397762c334f8fa4e234963829f905a664c7c48 Mon Sep 17 00:00:00 2001
**Previous From: Meticulus <theonejohnnyd@gmail.com>
**Previous Date: Fri, 1 Dec 2017 16:17:50 -0600
**Previous Subject: [PATCH] "0001-init-vendor-stubs-boot-critical
**_______________________________
**
***Previous From 41db2ec4e7477de3ac472470e3ef3f56a65c084e Mon Sep 17 00:00:00 2001
***Previous From: Meticulus <theonejohnnyd@gmail.com>
***Previous Date: Sat, 24 Dec 2016 23:43:28 -0600
***Previous Subject: [PATCH] 0001-init-vendor-stubs
***
***Change-Id: I31679a326358efa3962ed18bbc128e379fbe6c46"""
---
 init/property_service.cpp |  4 +++-
 init/vendor_init.cpp      | 12 +++++++++++-
 init/vendor_init.h        |  4 +++-
 3 files changed, 17 insertions(+), 3 deletions(-)

diff --git a/init/property_service.cpp b/init/property_service.cpp
index 49ba47e..7861510 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -684,9 +684,11 @@ void property_load_boot_defaults() {
     if (android::base::GetBoolProperty("ro.persistent_properties.ready", false)) {
         update_sys_usb_config();
     }
+    vendor_load_default_properties();
 }
 
 static void load_override_properties() {
+    vendor_load_persist_properties();
     if (ALLOW_LOCAL_PROP_OVERRIDE) {
         load_properties_from_file("/data/local.prop", NULL);
     }
@@ -753,7 +755,7 @@ void load_system_props() {
     load_properties_from_file("/factory/factory.prop", "ro.*");
 
     // Update with vendor-specific property runtime overrides
-    vendor_load_properties();
+    vendor_load_system_properties();
 
     load_recovery_id_prop();
 }
diff --git a/init/vendor_init.cpp b/init/vendor_init.cpp
index d3fd5ff..a30c2f1 100644
--- a/init/vendor_init.cpp
+++ b/init/vendor_init.cpp
@@ -32,6 +32,16 @@ IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 /* init vendor override stubs */
 
 __attribute__ ((weak))
-void vendor_load_properties()
+void vendor_load_default_properties()
+{
+}
+
+__attribute__ ((weak))
+void vendor_load_system_properties()
+{
+}
+
+__attribute__ ((weak))
+void vendor_load_persist_properties()
 {
 }
diff --git a/init/vendor_init.h b/init/vendor_init.h
index 9afb449..f86c65f 100644
--- a/init/vendor_init.h
+++ b/init/vendor_init.h
@@ -29,5 +29,7 @@ IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #ifndef __INIT_VENDOR__H__
 #define __INIT_VENDOR__H__
-extern void vendor_load_properties(void);
+extern void vendor_load_default_properties(void);
+extern void vendor_load_system_properties(void);
+extern void vendor_load_persist_properties(void);
 #endif /* __INIT_VENDOR__H__ */
-- 
2.7.4

